Jenkinsfile

```diff
@@ -1 +1,11 @@
-buildPlugin()
+/*
+ See the documentation for more options:
+ https://github.com/jenkins-infra/pipeline-library/
+*/
+buildPlugin(
+  forkCount: '1C', // Run 1 test process per core
+  useContainerAgent: true, // Set to `false` if you need to use Docker for containerized tests
+  configurations: [
+    [platform: 'linux', jdk: 21],
+    [platform: 'windows', jdk: 17],
```

Comment:
Can we have more configurations? I'd love to test it on Linux-JDK8 and Linux-JDK11, if possible.

Why? The OpenClover (org.openclover:clover) officially supports JDK7-9 only. In its latest 4.5.0 release an experimental support for JDK10-17 was added.

Reply:
Yes, you're welcome to test more configurations, though testing Java 8 won't work with this pull request because Jenkins 2.361.1 and later do not support Java 8.  Jenkins 2.361.1 and later use Java 11 byte code and support only Java 11 and Java 17.

Agents can still run builds with Java 8, but the controller itself must run with Java 11 or Java 17.  The agents also must run with Java 11 or Java 17 but they can control jobs that run any any JDK version or any other tool version.

Reply:
I didn't include a test of Java 11 in this case because the Java 17 compiler is configured by the Jenkins build tools to generate Java 11 byte code.  Likewise for the Java 21 compiler.  They are both generating and executing Java 11 byte code, so it did not seem especially useful to create a Java 11 test configuration to generate Java 11 byte code again.

I was intentionally accepting the risk of not running tests directly on Java 11 because the Jenkins project will end its support of Java 11 in Oct 2024 when Eclipse Temurin and other Java providers are expected to end their public support of Java 11.  We're only 12 months away from needing to remove that test configuration and the test configuration is already reasonably covered by the Java 17 and Java 21 configurations.

If you feel strongly that you want an explicit Java 11 configuration, I'm willing to add it.  I think it is testing things that are already well tested by other configurations, but I am less attached to reducing test configurations than I am to testing Java 21 in more plugins.